core/src/main/scala/cats/Unfoldable.scala

```diff
@@ -0,0 +1,59 @@
+package cats
+
+import simulacrum.typeclass
+
+@typeclass trait Unfoldable[F[_]] { self =>
+
+  /**
+    * Build an F[A] from a `seed` and a generating function `f``
+    * The generating function `f` behave as follows:
+    * - If `f(b)` is `None`, then `unfoldLeft(b)(f)` should be empty.
+    * - If `f(b)` is `Some((b1, a))`, then `unfoldLeft(b)(f)` should consist of `a` appended by `unfoldLeft(b1)(f)`
+    */
+  def unfoldLeft[A, B](seed: B)(f: B => Option[(B, A)]): F[A]
```

Comment:
In some cases, it could be useful to keep the `B` value at the end of the process, what do you think?